src/Module/Profile/UnkMail.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Profile;
+
+use Friendica\App;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Database\Database;
+use Friendica\Model\Mail;
+use Friendica\Model\User;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Unknown Mail module
+ */
+class UnkMail extends \Friendica\BaseModule
+{
+	/** @var IHandleUserSessions */
+	private $userSessions;
+
+	/** @var SystemMessages */
+	private $systemMessages;
+
+	/** @var Database */
+	private $database;
+
+	/** @var App\Page */
+	private $page;
+
+	public function __construct(App\Page $page, Database $database, SystemMessages $systemMessages, IHandleUserSessions $userSessions, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->userSessions   = $userSessions;
+		$this->systemMessages = $systemMessages;
+		$this->database       = $database;
+		$this->page           = $page;
+	}
+
+	protected function post(array $request = [])
+	{
+		$replyto = $this->userSessions->getMyUrl();
+		if (!$replyto) {
+			$this->systemMessages->addNotice($this->l10n->t('Permission denied.'));
+			return;
+		}
+
+		$recipient = $this->parameters['nickname'];
+		$subject   = trim($request['subject'] ?? '');
+		$body      = Strings::escapeHtml(trim($request['body'] ?? ''));
+
+		if (!$body) {
+			$this->systemMessages->addNotice($this->l10n->t('Empty message body.'));
+			return;
+		}
+
+		$user = User::getByNickname($recipient);
+		if (empty($user)) {
+			return;
+		}
+
+		if (!$user['unkmail']) {
+			return;
+		}
+
+		$total = $this->database->count('mail', ["`uid` = ? AND `created` > ? AND `unknown`", $user['uid'], DateTimeFormat::utc('now - 1 day')]);
```

Comment:
Maybe we can use a `Mail` model class instead of executing the dB directly?

Reply:
Maybe, but for now I'm trying to move module classes as fast as I can, and this is definitely more work than I think is necessary for this mostly unused feature.